.github/workflows/snyk_sca_scan.yml

```diff
@@ -0,0 +1,45 @@
+name: Snyk Code Security
+
+on:
+  pull_request:
+    branches:
+      - '**'
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '**/*.md'
+
+jobs:
+  open-source:
+    name: '🔒 Open Source Scan'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 16
+          cache: 'yarn'
+      - run: yarn
+      - run: yarn install
+
+      - uses: actions/checkout@v2
```

Comment:
Does snyk require a checkout? Does snyk require an install? I suspect you can remove all but one checkout.

Reply:
After discussion with @zzmp and @AndrewMohawk decided to include `--frozen-lockfile` to CI.yml workflow, ensuring the package.json has not been modified and dependencies are intact. Then in the Snyk workflow, we checkout and continue the scan.